app/lib/package/api_export/api_exporter.dart

```diff
@@ -216,6 +216,19 @@ final class ApiExporter {
     final versions = await packageBackend.tarballStorage
         .listVersionsInCanonicalBucket(package);
 
+    // Check versions that are not exposed in the public API and if they are moderated, delete them.
+    for (final cv in versions.entries) {
+      final version = cv.key;
+      if (versionListing.versions.any((v) => v.version == version)) {
+        continue;
+      }
+      final pv = await packageBackend.lookupPackageVersion(package, version);
+      if (pv != null && pv.isModerated) {
```

Comment:
We don't need this.

`versions` is the list of archives objects that exist in `tarballStorage`.

Later we do:
```
    // Remove versions that are not exposed in the public API.
    versions.removeWhere(
      (version, _) => !versionListing.versions.any((v) => v.version == version),
    );
```
which ensures that `versions` only contains entries that are explicitly listed `versionListing`.

And `synchronizeTarballs` will delete files not present in `versions`.

So we don't need to delete files, the only case where `synchronizeTarballs` doesn't delete an archive not in `versions`, is if the object was recently modified.
This is to avoid race conditions, but recently modified is 3 hours -- we could set that lower if we're at all concerned here.